designs/usb-light.stanza

```diff
@@ -31,15 +43,75 @@ pcb-module main-board:
   val vbus-voltage = (nom-value(recommended-voltage(property(conn.VCC.power-pin)))) as Double
   val exact-resistance = led-circuit-resistance(vbus-voltage, property(led[0].forward-voltage), property(led[0].test-current), 2)
   inst res : gen-res-cmp(closest-std-val(exact-resistance,5.0), "0603")[4]
+  val trace-thickness = 0.254
   for i in 0 to 4 do :
+    ; place parts
+    val led-i4-point = Point(-2.5, 5.0 * to-double(i) - 7.5)
+    place(led[i + 4]) at loc(led-i4-point, 90.0) on Top
+    val led-i-point = Point(-7.5, 5.0 * to-double(i) - 7.5)
+    place(led[i]) at loc(led-i-point, 90.0) on Top
+    val res-i-point = Point(2.5, 5.0 * to-double(i) - 7.5)
+    place(res[i]) at loc(res-i-point, 90.0) on Top
+    ; assign nets
     net (p5v led[i].a)
-    net (led[i + 4].a led[i].c)
-    net (res[i].p[1] led[i + 4].c)
+
+    val net-name1 = string-join(["net-led", i + 1])
+    val net-id1 = make-net( to-symbol(net-name1), [(led[i + 4].a)])
+    net (net-id1 led[i].c)
+
+    val net-name2 = string-join(["net-res", i + 1])
+    val net-id2 = make-net( to-symbol(net-name2), [(res[i].p[1])])
+    net (net-id2 led[i + 4].c)
+
     net (gnd res[i].p[2])
-    place(led[i + 4]) at loc(-2.5, 5.0 * to-double(i) - 7.5, 90.0) on Top
-    place(led[i]) at loc(-7.5, 5.0 * to-double(i) - 7.5, 90.0) on Top
-    place(res[i]) at loc(2.5, 5.0 * to-double(i) - 7.5, 90.0) on Top
-  place(conn) at loc(10.75, 0.0, 90.0) on Top
+    ; place p5v track stub
+    val led-5v-x1 = x(led-i-point) - y(center(get-pad-pose(((led[i])), "a")))
```

Comment:
This (and basically every point computed below) looks like it's subtracting an x coordinate from a y coordinate. This is usually an indicator that something is wrong. I'm assuming this design works, but I'm suspicious that this would break if you did something like change the spacing between rows. Do you know why you need to mix coordinates here?

Reply:
Yea this looks a little funny. get-pad-pose returns the pad pose in relation to the center of the component. Here, the components are placed rotated 90 degrees. So to find the shift X in relation to the pad to the board, I take the shift y of the pad to the component then apply that to the x of the component to the board. I think these kinds of functions highlight an opportunity to build higher order functions for this type of behavior.